prow/test/integration/setup-prow-components.sh

```diff
@@ -135,33 +187,28 @@ function create_prowimagebuilder_yaml() {
 }
 
 function remove_old_components() {
-  declare -a prow_components
-  local prow_component
+  declare -a components
+  local component
   if (($#)); then
     log "Removing select Prow components"
-    for image in "$@"; do
-      if [[ -v "PROW_IMAGES_TO_COMPONENTS[${image}]" ]]; then
-        for prow_component in ${PROW_IMAGES_TO_COMPONENTS[${image}]//,/ }; do
-          prow_components+=("${prow_component}")
-        done
-      fi
+    for component in "$@"; do
+      components+=("${component}")
     done
-
   else
     log "Removing *all* Prow components (if any)"
-    for prow_component in "${PROW_COMPONENTS[@]}"; do
-      prow_components+=("${prow_component}")
+    for component in "${PROW_COMPONENTS[@]}"; do
+      components+=("${component}")
     done
   fi
 
-  for prow_component in "${prow_components[@]}"; do
-    do_kubectl delete deployment -l app="${prow_component}"
-    do_kubectl delete pods -l app="${prow_component}"
+  for component in "${components[@]}"; do
+    do_kubectl delete deployment -l app="${component}"
```

Comment:
IIRC we should not have to delete old deplyments/pods. Doesn't kubectl apply just apply the new changes to the cluster? Or am I misunderstanding?

Reply:
Thinking about this some more, maybe it's better to delete the old deployments/pods, because it guarantees that all state is lost (that the new deployments will start out fresh --- e.g., any in-memory caches in the binaries running in the containers will be empty). I suppose there might be a case where we want to only `kubectl apply` without deleting the old stuff, but that should belong in a separate flag. WDYT?

Reply:
> aligned with what we actually do in production.

Good point. Let me push up a new change.

Reply:
Done. Updated the commit message. To be clear, the old behavior used `kubectl delete ...` and then `kubectl apply ...` for deploying components by default. The new behavior is to only use `kubectl apply` by default, and to move the `kubectl delete` operations behind the new `-delete` flag. I didn't want to completely remove this (now optional) functionality because I couldn't think of a strong enough reason to do so.